### DISCUSSION TITLE
Recipe terminated with fatal error: spawn latexmk ENOENT.

### DISCUSSION THREAD

**Discussion by jossiej00** [James-Yu/LaTeX-Workshop#3094] [Q&A] [answered]
### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you still see this issue. [Required]

You still see this issue?: Yes

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before filling an issue.

You visited the wiki?: Yes

### If your issue is with compiling a document (not having to do with finding the root file of a project), check first that you can compile manually.

You can compile a TeX document manually?: No

## Describe the bug [Required]

Below is the log copied. And the warning showed "Recipe terminated with fatal error: spawn latexmk ENOENT." as well as "chktex could not be found. after running a latex file.

### To Reproduce

1. clone a project from github
2. open the latex file stored in this project
3. install "LaTex-Workshop" extension
4. run it
5. and warnings come up

### Expected behavior

should compile successfully and create a related pdf file.

## Logs [Required]

Please paste the whole log messages here, not parts of ones. The log should start with `Initializing LaTeX Workshop`. It is very important to identify problems.

### LaTeX Workshop Output [Required]

[17:01:07] Initializing LaTeX Workshop.
[17:01:07] Extension root: /Users/jiangyuxin/.vscode/extensions/james-yu.latex-workshop-8.23.0
[17:01:07] $PATH: /Library/Frameworks/Python.framework/Versions/3.10/bin:/usr/local/mysql/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/Users/jiangyuxin/anaconda3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
[17:01:07] $SHELL: /bin/bash
[17:01:07] vscode.env.appName: Visual Studio Code
[17:01:07] vscode.env.remoteName: undefined
[17:01:07] vscode.env.uiKind: 1
[17:01:07] editor.acceptSuggestionOnEnter: "on"
[17:01:07] latex-workshop.bind.enter.key: true
[17:01:07] latex-workshop.docker.enabled: false
[17:01:07] latex-workshop.docker.image.latex: ""
[17:01:07] latex-workshop.hover.preview.mathjax.extensions: []
[17:01:07] latex-workshop.intellisense.package.enabled: true
[17:01:07] latex-workshop.intellisense.update.aggressive.enabled: false
[17:01:07] latex-workshop.intellisense.update.delay: 1000
[17:01:07] latex-workshop.latex.autoBuild.run: "onFileChange"
[17:01:07] latex-workshop.latex.outDir: "%DIR%"
[17:01:07] latex-workshop.latex.recipes: [
 {
  "name": "latexmk 🔃",
  "tools": [
   "latexmk"
  ]
 },
 {
  "name": "latexmk (latexmkrc)",
  "tools": [
   "latexmk_rconly"
  ]
 },
 {
  "name": "latexmk (lualatex)",
  "tools": [
   "lualatexmk"
  ]
 },
 {
  "name": "pdflatex ➞ bibtex ➞ pdflatex × 2",
  "tools": [
   "pdflatex",
   "bibtex",
   "pdflatex",
   "pdflatex"
  ]
 },
 {
  "name": "Compile Rnw files",
  "tools": [
   "rnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "Compile Jnw files",
  "tools": [
   "jnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "tectonic",
  "tools": [
   "tectonic"
  ]
 }
]
[17:01:07] latex-workshop.latex.tools: [
 {
  "name": "latexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-pdf",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "lualatexmk",
  "command": "latexmk",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-lualatex",
   "-outdir=%OUTDIR%",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "latexmk_rconly",
  "command": "latexmk",
  "args": [
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "pdflatex",
  "command": "pdflatex",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "%DOC%"
  ],
  "env": {}
 },
 {
  "name": "bibtex",
  "command": "bibtex",
  "args": [
   "%DOCFILE%"
  ],
  "env": {}
 },
 {
  "name": "rnw2tex",
  "command": "Rscript",
  "args": [
   "-e",
   "knitr::opts_knit$set(concordance = TRUE); knitr::knit('%DOCFILE_EXT%')"
  ],
  "env": {}
 },
 {
  "name": "jnw2tex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"tex\")"
  ],
  "env": {}
 },
 {
  "name": "jnw2texmintex",
  "command": "julia",
  "args": [
   "-e",
   "using Weave; weave(\"%DOC_EXT%\", doctype=\"texminted\")"
  ],
  "env": {}
 },
 {
  "name": "tectonic",
  "command": "tectonic",
  "args": [
   "--synctex",
   "--keep-logs",
   "%DOC%.tex"
  ],
  "env": {}
 }
]
[17:01:07] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto"
[17:01:07] Creating a new file watcher.
[17:01:07] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[17:01:07] Creating PDF file watcher.
[17:01:07] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{}}
[17:01:07] Creating Bib file watcher.
[17:01:07] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[17:01:07] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[17:01:07] Cannot run pdflatex to determine if we are using MiKTeX
[17:01:07] [Server] Creating LaTeX Workshop http and websocket server.
[17:01:07] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[17:01:07] LaTeX Workshop initialized.
[17:01:07] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[17:01:07] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[17:01:07] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[17:01:07] Current workspace folders: undefined
[17:01:07] Current workspaceRootDir: 
[17:01:07] [Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":51943}
[17:01:07] Snippet data loaded.
[17:01:11] Current workspace folders: undefined
[17:01:11] Current workspaceRootDir: 
[17:01:11] Found root file from active editor: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:11] Root file changed: from undefined to /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:11] Start to find all dependencies.
[17:01:11] Root file languageId: latex
[17:01:11] Reset file watcher.
[17:01:11] Parsing a file and its subfiles: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:11] Parse fls file.
[17:01:11] Cannot find fls file: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.fls
[17:01:11] Added to file watcher: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:11] Checking for duplicate labels: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex.
[17:01:13] Manager.fileWatcher.getWatched: {"/Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1":["hw1.tex"]}
[17:01:13] Manager.filesWatched: ["/Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex"]
[17:01:13] BibWatcher.bibWatcher.getWatched: {}
[17:01:13] BibWatcher.bibsWatched: []
[17:01:13] PdfWatcher.pdfWatcher.getWatched: {}
[17:01:13] PdfWatcher.pdfsWatched: []
[17:01:20] BUILD command invoked.
[17:01:20] The document of the active editor: file:///Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:20] The languageId of the document: latex
[17:01:20] Current workspace folders: undefined
[17:01:20] Current workspaceRootDir: 
[17:01:20] Found root file from active editor: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:20] Keep using the same root file: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:20] Building root file: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:20] Build root file /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex
[17:01:20] outDir: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1
[17:01:20] Recipe step 1: latexmk, -synctex=1,-interaction=nonstopmode,-file-line-error,-pdf,-outdir=/Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1,/Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1
[17:01:20] Recipe step env: {}
[17:01:20] cwd: /Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1
[17:01:20] LaTeX build process spawned. PID: undefined.
[17:01:20] LaTeX fatal error: spawn latexmk ENOENT, . PID: undefined.
[17:01:20] Does the executable exist? $PATH: /Library/Frameworks/Python.framework/Versions/3.10/bin:/usr/local/mysql/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/Users/jiangyuxin/anaconda3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
[17:01:20] Does the executable exist? $Path: undefined
[17:01:20] The environment variable $SHELL: /bin/bash
[17:01:22] Manager.fileWatcher.getWatched: {"/Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1":["hw1.tex"]}
[17:01:22] Manager.filesWatched: ["/Users/jiangyuxin/Documents/jossie/MATH303/HW/HW1/hw1.tex"]
[17:01:22] BibWatcher.bibWatcher.getWatched: {}
[17:01:22] BibWatcher.bibsWatched: []
[17:01:22] PdfWatcher.pdfWatcher.getWatched: {}
[17:01:22] PdfWatcher.pdfsWatched: []

### Developer Tools Console [Required]

```
To access the log, click 'help' -> 'Toggle Developer Tools' -> 'Console'. Paste anything suspicious.

[Paste the log here]
```

## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: MacMonterey 12.1
 - VS Code version: 1.63.2 (Universal)
 - Extension version: v8.23.0
 - TeX distribution version: MikTex Console 4.5

### Are you using VSCodium?

No

### Are you using the Snap or Flatpack versions of VS Code?

No

### Are you using LaTeX Workshop with VS Code Remote?

No

If the answer is Yes, please write which one you are using. Write the versions of the remote extension.

- Remote Containers/Remote SSH/Remote WSL
- Remote Extension Version: [e.g. 0.48.0]

**Comment by tamuratak** [comment]
> If your issue is with compiling a document (not having to do with finding the root file of a project), check first that you can compile manually.
>
> You can compile a TeX document manually?: No

This issue is not related to LaTeX Workshop. Please find help somewhere else.

> TeX distribution version: MikTex Console 4.5

We don't know much about MiKTeX. We strongly recommend users to install TeX Live.

Please note MiKTeX does not ship with Perl. You have to install Perl by yourself, which latexmk requires.

See https://github.com/James-Yu/LaTeX-Workshop/wiki/Install#requirements

**Comment by James-Yu** [comment, marked as answer]
You do not have the TeX binaries in your path:
```
Does the executable exist? $PATH: /Library/Frameworks/Python.framework/Versions/3.10/bin:/usr/local/mysql/bin:/Library/Frameworks/Python.framework/Versions/3.6/bin:/Users/jiangyuxin/anaconda3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin
```
Include your TeX installation in PATH. Please refer to your TeX distribution.
